### PR TITLE
Filter plugin

### PR DESCRIPTION
**Pull request by vladminsky**

### CODE REVIEW

File: plugins/legend.js [hunk 1]
```diff
@@ -252,6 +251,16 @@
                                 _.intersection(colorScaleConfig.order, domain),
                                 domain
                             );
+                        } else {
+                            var orderState = self._legendOrderState[c];
+                            domain = domain.sort(function (a, b) {
```
Comment (Mavrin) [inline plugins/legend.js]: Please simplify. Or replace on `if`

File: plugins/quick-filter.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+                    (xSettings) :
+                    (_(sources.dims).keys()));
+
+                var chartData = self._chart.getData();
+
+                self._fields
+                    .filter(function (dim) {
+                        var isMeasure = (sources.dims[dim].type === 'measure');
+                        if (!isMeasure) {
+                            console.log('The [' + dim + '] isn\'t measure so Quick Filter plugin skipped it');
```
Comment (Mavrin) [inline plugins/quick-filter.js]: Use logger from chart.

File: plugins/quick-filter.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+                this._chart.removeFilter(filterId);
+
+                state[dim] = this._chart.addFilter({
+                    tag: 'quick-filter',
+                    predicate: function (row) {
+                        return !isRowMatch(row);
+                    }
+                });
+
+                this._chart.refresh();
```
Comment (Mavrin) [inline plugins/quick-filter.js]: It looks strange and makes data of inconsistent.

Comment (vladminsky) [inline plugins/quick-filter.js]: Let's introduce a better solution in a separate PR

**Pull request merged**